lib/matplotlib/axes/_secondary_axes.py

```diff
@@ -145,10 +146,27 @@ def apply_aspect(self, position=None):
         self._set_lims()
         super().apply_aspect(position)
 
-    @_docstring.copy(Axis.set_ticks)
-    def set_ticks(self, ticks, labels=None, *, minor=False, **kwargs):
-        ret = self._axis.set_ticks(ticks, labels, minor=minor, **kwargs)
-        self.stale = True
+    @functools.wraps(_AxesBase.set_xticks)
+    def set_xticks(self, *args, **kwargs):
+        if self._orientation == "y":
+            raise TypeError("Cannot set xticks on a secondary y-axis")
+        ret = super().set_xticks(*args, **kwargs)
+        if self._orientation == "x":
+            self._ticks_set = True
+        return ret
```

Comment:
Why both types of `if` checks? Are you expecting this to occur on 3D or some other `Axes` type?

Reply:
Because the second one came in first and I forgot to remove it after adding the first one.  Fixed.